### PR TITLE
Fixed hidden path-not-found diagnostic

### PR DESCRIPTION
**Pull request by yuvalsw**
This change is [Reviewable](https://reviewable.io/reviews/starkware-libs/cairo/1266)

### CODE REVIEW

**Review by orizi — APPROVED** [review]
:lgtm:

> Reviewed 3 of 3 files at r1, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/starkware-libs/cairo/1266#-:-NJ9_R9SAEtd5ZNGBFJ-:bnfp4nl)* status: all files reviewed, 2 unresolved discussions (waiting on @spapinistarkware and @yuvalsw)

___
*[`crates/semantic/src/resolve_path.rs` line 494 at r1](https://reviewable.io/reviews/starkware-libs/cairo/1266#-NJ9_8A78OEfODyvru3R:-NJ9_8A78OEfODyvru3S:bfyx9j6) ([raw file](https://github.com/starkware-libs/cairo/blob/053f6ac4f49227946690448967fa9448c8d32da1/crates/semantic/src/resolve_path.rs#L494)):*
> ```XML
>             ModuleItemId::Submodule(id) => ResolvedGenericItem::Module(ModuleId::Submodule(id)),
>             ModuleItemId::Use(id) => {
>                 // TODO(spapini): Right now we call priv_use_semantic_data() directly for cycle
> ```

Consider how to rephrase this todo - or remove it if relevant now.



_Code quote:_
```XML
                // TODO(spapini): Right now we call priv_use_semantic_data() directly for cycle
                // handling. Otherwise, we need to handle cycle both on it and on the selector
                // use_resolved_item(). Fix this.
```
___
*[`crates/semantic/src/resolve_path.rs` line 497 at r1](https://reviewable.io/reviews/starkware-libs/cairo/1266#-NJ9_MQOEFSdieme2qsd:-NJ9_MQOEFSdieme2qse:bq38m23) ([raw file](https://github.com/starkware-libs/cairo/blob/053f6ac4f49227946690448967fa9448c8d32da1/crates/semantic/src/resolve_path.rs#L497)):*
> ```XML
>                 // handling. Otherwise, we need to handle cycle both on it and on the selector
>                 // use_resolved_item(). Fix this.
>                 // TODO(yg): other places?
> ```

improve phrasing.



_Code quote:_
```XML
// TODO(yg): other places?
```

**Review by yuvalsw — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/starkware-libs/cairo/1266)* status: all files reviewed, 2 unresolved discussions (waiting on @orizi and @spapinistarkware)

___
*[`crates/semantic/src/resolve_path.rs` line 494 at r1](https://reviewable.io/reviews/starkware-libs/cairo/1266#-NJ9_8A78OEfODyvru3R:-NJAOeE1BpaKlmXXw2jf:b-896fix) ([raw file](https://github.com/starkware-libs/cairo/blob/053f6ac4f49227946690448967fa9448c8d32da1/crates/semantic/src/resolve_path.rs#L494)):*
<details><summary><i>Previously, orizi wrote…</i></summary><blockquote>

Consider how to rephrase this todo - or remove it if relevant now.
</blockquote></details>

Done.
___
*[`crates/semantic/src/resolve_path.rs` line 497 at r1](https://reviewable.io/reviews/starkware-libs/cairo/1266#-NJ9_MQOEFSdieme2qsd:-NJAMqKqCWMNfcRkhOYD:b-896fix) ([raw file](https://github.com/starkware-libs/cairo/blob/053f6ac4f49227946690448967fa9448c8d32da1/crates/semantic/src/resolve_path.rs#L497)):*
<details><summary><i>Previously, orizi wrote…</i></summary><blockquote>

improve phrasing.
</blockquote></details>

Done.

**Review by orizi — APPROVED** [review]
> Reviewed 2 of 2 files at r2, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/starkware-libs/cairo/1266)* status: :shipit: complete! all files reviewed, all discussions resolved (waiting on @spapinistarkware)
